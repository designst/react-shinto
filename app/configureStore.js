/* @flow */

import { createLogger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import { routerMiddleware } from 'react-router-redux';
import { compose, createStore, applyMiddleware } from 'redux';

// import type { Store } from '../types';
import createReducer from './createReducer';

const sagaMiddleware = createSagaMiddleware();

export default (history, initialState) => {
  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [
    sagaMiddleware,
    routerMiddleware(history),
  ];

  let composeEnhancers = compose;

  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    const loggerMiddleware = createLogger();
    middlewares.push(loggerMiddleware);

    // If Redux DevTools Extension is installed use it, otherwise use Redux compose
    /* eslint-disable no-underscore-dangle */
    // noinspection JSUnresolvedVariable
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
      // noinspection JSUnresolvedFunction
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        // Prevent recomputing reducers for `replaceReducer`
        shouldHotReload: false,
      });
    }
    /* eslint-enable */
  }

  const enhancers = [
    applyMiddleware(...middlewares),
  ];

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers)
  );

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectSagas = {}; // Saga registry
  store.injectReducers = {}; // Reducer registry

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./createReducer', () => {
      try {
        const nextReducer = require('./createReducer').default;

        store.replaceReducer(nextReducer);
      } catch (error) {
        console.error(`==> ðŸ˜­  Reducer hot reloading error ${error}`);
      }
    });
  }

  return store;
};
