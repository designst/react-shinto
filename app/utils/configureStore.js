/* @flow */

import fp from 'lodash/fp';
import thunk from 'redux-thunk';
import { init } from '@rematch/core';
import { createLogger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import createLoadingPlugin from '@rematch/loading';
import { routerMiddleware } from 'react-router-redux';
import createWaitForActionMiddleware from 'redux-wait-for-action';

import createReducer, { rootReducers } from './createReducer';
import createApiService, { createApiServicePlugin } from './apiService';

const loadingPluginOptions = {};
const loadingPlugin = createLoadingPlugin(loadingPluginOptions);

export const createFakeReducers = initialState => {
  const fakeReducer = (state = {}) => state;
  const rootReducerKeys = Object.keys(rootReducers);

  return Object.keys(initialState)
    .filter(key => !rootReducerKeys.includes(key))
    .reduce((reducers, key) => {
      reducers[key] = fakeReducer;
      return reducers;
    }, {});
};

export default (history, initialState, configuration = {}) => {
  // Create the store with three middlewares
  // 1. thunkMiddleware: Makes thunk work
  // 2. sagaMiddleware: Makes redux-sagas work
  // 3. routerMiddleware: Syncs the location/URL path to the state

  const { auth: { token } = {} } = initialState;
  const apiService = createApiService(token);

  const sagaMiddleware = createSagaMiddleware({
    context: {
      apiService,
    },
  });

  const waitForActionMiddleware = createWaitForActionMiddleware();

  const middlewares = [thunk, sagaMiddleware, waitForActionMiddleware, routerMiddleware(history)];

  if (__DEV__ && __CLIENT__) {
    const loggerMiddleware = createLogger();
    middlewares.push(loggerMiddleware);
  }

  // Create fake reducers which are not already known by the injectedReducers map
  // but which are expected by the server-side rendered initialState object.
  // The created fakeReducers will be replaced after client-side reducers are injected.
  const initialStateReducers = createFakeReducers(initialState);

  const store = init(
    fp.assign(
      {
        redux: {
          middlewares,
          initialState,
          reducers: rootReducers,
          combineReducers: rematchReducers => {
            const { injectedReducers = {} } = store || {};
            // Override rematch combineReducers to handle the fakeReducers which are reducers
            // not already known but expected by the initialState object and to handle the
            // already injectedReducers.

            return createReducer(
              // injected modelReducers are already created by the rematch model function,
              // so we can ignore them here to avoid additional loops.
              {},
              {
                ...rematchReducers,
                ...injectedReducers,
              },
              initialStateReducers,
            );
          },
        },
        plugins: [loadingPlugin, createApiServicePlugin(apiService)],
      },
      configuration,
    ),
  );

  apiService.store = store;

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectedSagas = {}; // Saga registry
  store.injectedModels = {}; // Model registry
  store.injectedReducers = {}; // Reducer registry
  store.initialStateReducers = initialStateReducers;

  /* istanbul ignore next */
  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./createReducer', () => {
      try {
        const createNextReducer = require('./createReducer').default;
        const nextReducer = createNextReducer(
          store.injectedModels,
          store.injectedReducers,
          store.initialStateReducers,
        );

        store.replaceReducer(nextReducer);
      } catch (error) {
        console.error(`==> Reducer hot reloading error ${error}`);
      }
    });
  }

  return store;
};
