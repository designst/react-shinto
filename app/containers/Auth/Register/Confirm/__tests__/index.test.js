import React from 'react';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';
import { browserHistory, StaticRouter } from 'react-router-dom';

import configureStore from 'utils/configureStore';

import ConnectedRegisterConfirm from '../index';

describe('<RegisterConfirm />', () => {
  const mockData = {
    match: {
      params: {
        token: 'token',
      },
    },
  };

  const store = configureStore(browserHistory, {});
  const tree = renderer
    .create(
      <Provider store={store}>
        <IntlProvider locale="en">
          <StaticRouter context={{}}>
            <ConnectedRegisterConfirm {...mockData} />
          </StaticRouter>
        </IntlProvider>
      </Provider>,
    )
    .toJSON();

  test('renders', () => {
    expect(tree).toMatchSnapshot();
  });
});
