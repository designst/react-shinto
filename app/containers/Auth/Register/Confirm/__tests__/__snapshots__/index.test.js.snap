// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterConfirm /> renders 1`] = `
<div
  className="b-register-confirm"
>
  <h1
    className="MuiTypography-root-1 MuiTypography-headline-6"
  >
    <span>
      Confirm your registration
    </span>
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root-25 MuiFormControl-marginNormal-26 MuiFormControl-fullWidth-28"
      onBlur={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-34 MuiInputLabel-root-29 MuiInputLabel-formControl-30 MuiInputLabel-animated-33 MuiInputLabel-shrink-32"
        data-shrink={true}
      >
        Token
      </label>
      <div
        className="MuiInput-root-39 MuiInput-fullWidth-46 MuiInput-formControl-40 MuiInput-underline-43"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoFocus={true}
          className="MuiInput-input-47"
          disabled={false}
          name="token"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Token"
          type="text"
          value="token"
        />
      </div>
      <p
        className="MuiFormHelperText-root-52"
      >
        Please enter your token
      </p>
    </div>
    <button
      className="MuiButtonBase-root-78 MuiButton-root-56 MuiButton-contained-65 MuiButton-containedPrimary-66 MuiButton-raised-68 MuiButton-raisedPrimary-69"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-57"
      >
        <span>
          Register
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-81"
      />
    </button>
  </form>
</div>
`;
